File: 백준/Bronze/25304. 영수증/영수증.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int totalAmount = Integer.parseInt(br.readLine());
        int goodsCount = Integer.parseInt(br.readLine());
        int totalPrice = 0;
        StringTokenizer st;
        for (int i = 0; i < goodsCount; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int price = Integer.parseInt(st.nextToken());
            int amount = Integer.parseInt(st.nextToken());
            totalPrice += price * amount;
        }
        System.out.println(totalAmount == totalPrice ? "Yes" : "No");
    }
}

File: 백준/Bronze/25304. 영수증/영수증.cc
#include <iostream>
using namespace std;

int main()
{
    int buyPrice;
    int totalPrice = 0;
    int productAmount;
    cin >> buyPrice >> productAmount;

    for (int i = 0; i < productAmount; i++)
    {
        int price, count;
        cin >> price >> count;
        totalPrice += price * count;
    }
    if (totalPrice == buyPrice) cout << "Yes";
    else cout << "No";
}